src/libcore/num/mod.rs

```diff
@@ -2504,76 +2504,172 @@ impl fmt::Display for TryFromIntError {
     }
 }
 
-macro_rules! same_sign_try_from_int_impl {
-    ($storage:ty, $target:ty, $($source:ty),*) => {$(
+// no possible bounds violation
+macro_rules! try_from_unbounded {
+    ($source:ty, $($target:ty),*) => {$(
         #[unstable(feature = "try_from", issue = "33417")]
         impl TryFrom<$source> for $target {
             type Error = TryFromIntError;
```

Comment:
Oh-- but I see there are cfg'd impls of this for `usize` based on the current target's pointer width, so those would have to be kept as `TryFromIntError` in order to ensure target compatibility.